tests/setup_test.go

```diff
@@ -2401,3 +2428,510 @@ func TestUpdateOrganization(t *testing.T) {
 		t.Fatalf("org mismatch (-want +got):\n%s", diff)
 	}
 }
+
+func setupTestOrgInvitation(t *testing.T, ctx context.Context, dir string, orgInvitationEnabled bool) (*config.Config, string, string) {
+	var c *config.Config
+	if orgInvitationEnabled {
+		_, c = setup(ctx, t, dir, withOrganizationMemberAddingMode(config.OrganizationMemberAddingModeInvitation))
+	} else {
+		_, c = setup(ctx, t, dir)
+	}
+
+	gwAdminClient := gwclient.NewClient(c.Gateway.APIExposedURL, c.Gateway.AdminToken)
+
+	_, _, err := gwAdminClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: agolaUser01})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	tokenUser01, _, err := gwAdminClient.CreateUserToken(ctx, agolaUser01, &gwapitypes.CreateUserTokenRequest{TokenName: "test"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, _, err = gwAdminClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: "user02"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	tokenUser02, _, err := gwAdminClient.CreateUserToken(ctx, "user02", &gwapitypes.CreateUserTokenRequest{TokenName: "test"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, _, err = gwAdminClient.CreateOrg(ctx, &gwapitypes.CreateOrgRequest{Name: "org01", Visibility: gwapitypes.VisibilityPublic})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	_, _, err = gwAdminClient.AddOrgMember(ctx, "org01", "user01", gwapitypes.MemberRoleOwner)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	return c, tokenUser01.Token, tokenUser02.Token
+}
+
+func TestOrgInvitation(t *testing.T) {
+	t.Run("test create org invitation", func(t *testing.T) {
+		dir := t.TempDir()
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		c, tokenUser01, _ := setupTestOrgInvitation(t, ctx, dir, true)
+		gwClientUser01 := gwclient.NewClient(c.Gateway.APIExposedURL, tokenUser01)
```

Comment:
To avoid this duplicated code you should use a test matrix like done in other tests containing the test name and test function and in a loop do the setup and call the test function.